packages/currency/src/types.ts

```diff
@@ -43,22 +48,28 @@ export type ERC777Currency = {
 /**
  * The minimum properties of a Currency
  */
-export type CurrencyInput =
-  | ({ type: RequestLogicTypes.CURRENCY.ETH | RequestLogicTypes.CURRENCY.BTC } & NativeCurrency)
-  | ({ type: RequestLogicTypes.CURRENCY.ISO4217 } & ISO4217Currency)
-  | ({ type: RequestLogicTypes.CURRENCY.ERC20 } & ERC20Currency)
-  | ({ type: RequestLogicTypes.CURRENCY.ERC777 } & ERC777Currency);
+export type CurrencyInput = NativeCurrency | ISO4217Currency | ERC20Currency | ERC777Currency;
 
 /**
  * The description of Currency, its core properties and some computed properties.
  * `meta` enables applications to add any metadata they need to a Currency
  */
-export type CurrencyDefinition<TMeta = unknown> = CurrencyInput & {
+export type FromInput<T, TMeta> = T & {
```

Comment:
I don't really understand the naming `FromInput`, what input is it? Isn't the comment outdated? Is the export still required?